### PR TITLE
Fallback to suggesting all files in _filedir_xspec

### PR DESCRIPTION
**Pull request by gftg85**

### CODE REVIEW

File: bash_completion [hunk 1]
```diff
@@ -1938,6 +1938,13 @@ _filedir_xspec()
         }
         ))
 
+    # Try without filter if it failed to produce anything and configured to
+    [[ -n ${COMP_FILEDIR_FALLBACK:-} && ${#toks[@]} -lt 1 ]] && {
+        reset=$(shopt -po noglob); set -o noglob
```
Comment (scop) [inline bash_completion]: Need local for reset.

File: bash_completion [hunk 2]
```diff
@@ -1938,6 +1938,13 @@ _filedir_xspec()
...
+        toks+=( $( eval compgen -f -- "\$(quote_readline "\$cur")" ) )
```
Comment (scop) [inline bash_completion]: Can we get away without the eval here? I.e. maybe just:
```
        toks+=( $( compgen -f -- "$(quote_readline "$cur")" ) )
```

**Comment by gftg85** [comment]
I applied your suggestions and tested that they still produce the output describe in the commit message.

Thanks!

File: bash_completion [hunk 3]
```diff
@@ -1904,7 +1904,7 @@ else
 fi
 _filedir_xspec()
 {
-    local cur prev words cword
+    local cur prev words cword local
```
Comment (scop) [inline bash_completion]: Um no, this is not correct :) Please remove this change.

File: bash_completion (hunk 1 above)
Comment (scop) [reply bash_completion]: Just add `local` before `reset` right here on this line and it's all good.

**Comment by gftg85** [comment]
Thanks, I have submitted an updated version.

**Pull request merged**